src/app/common/common.tsx

```diff
@@ -784,14 +785,78 @@ function HeaderRenderer(props: any, hnum: number): any {
     return <div className={cn("title", "is-" + hnum)}>{props.children}</div>;
 }
 
-function CodeRenderer(props: any): any {
-    return <code className={cn({ inline: props.inline })}>{props.children}</code>;
-}
-
 @mobxReact.observer
-class Markdown extends React.Component<{ text: string; style?: any; extraClassName?: string }, {}> {
+class Markdown extends React.Component<{ text: string; style?: any; extraClassName?: string; codeSelect?: boolean}, {}> {
+    codeBlockList: Array<React.RefObject<HTMLElement>>    
+    codeSelectMarkdownID: number;
+
+    constructor(props: any) {
+        super(props);  
+        this.codeBlockList = [];
+        if(this.props.codeSelect) {
+            let model = GlobalModel;
+            let inputModel = model.inputModel; 
+            this.codeSelectMarkdownID = inputModel.registerCodeSelectMarkdown();   
+        }
+    }  
+
+   CodeRenderer(props: any, codeSelect: boolean): any { 
```

Comment:
I moved this inside the markdown class, is that okay?

The pattern is kind of off now, but seems like functionality wise no one else was depending on this